### PR TITLE
fix: prevent quit_on_open from opening file in the same window

### LINKED ISSUE #1628: File opens on the same nvim-tree window when quit_on_open = true
### Description

When opening a file using Telescope or :e [filename] while focusing on the nvim-tree window, the file opens the same window as nvim-tree

https://user-images.githubusercontent.com/17254073/194481452-6e013b10-c9b9-461d-872f-25c06bc40928.mp4

### Neovim version

```text
NVIM v0.9.0-dev-1214-ge54541f7f-dirty
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

MacOS 12.6

### nvim-tree version

main - 7282f7d

### Minimal config

```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "kyazdani42/nvim-tree.lua",
      "kyazdani42/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
    actions = {
      open_file = {
      quit_on_open = true
    }
  }
}
end
```

### Steps to reproduce

1- nvim -nu /tmp/nvt-min.lua
2- e ./nvt-min.lua
3- :NvimTreeOpen
4- e ./nvt-min.lua

### Expected behavior

_No response_

### Actual behavior

_No response_

### PR DESCRIPTION
**Pull request by Almo7aya**
fixes #1628

 call `view._prevent_buffer_override` on `BufWipeout` instead of `view.abandon_current_window` when `quit_on_open = true`

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -373,7 +373,7 @@ local function setup_autocommands(opts)
   if not opts.actions.open_file.quit_on_open then
     create_nvim_tree_autocmd("BufWipeout", { pattern = "NvimTree_*", callback = view._prevent_buffer_override })
   else
-    create_nvim_tree_autocmd("BufWipeout", { pattern = "NvimTree_*", callback = view.abandon_current_window })
+    create_nvim_tree_autocmd("BufWipeout", { pattern = "NvimTree_*", callback = view._prevent_buffer_override })
```
Comment (alex-courtis) [inline lua/nvim-tree.lua:376]: remove if; both branches are the same

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Tested OK: https://github.com/nvim-tree/nvim-tree.lua/issues/1628#issuecomment-1272185627

**Comment by alex-courtis** [comment]
Also resolves an issue when loading a session with `quit_on_open`.

```
Error detected while processing BufWipeout Autocommands for "NvimTree_*":
Error executing lua callback: ...ck/packer/start/nvim-tree.lua.dev/lua/nvim-tree/view.lua:312: attempt to index a nil value
stack traceback:
        ...ck/packer/start/nvim-tree.lua.dev/lua/nvim-tree/view.lua:312: in function <...ck/packer/start/nvim-tree.lua.dev/lua/nvim-tree/view.lua:309>
        [C]: at 0x5648579a27f0
        [C]: in function 'pcall'
        ...ck/packer/start/nvim-tree.lua.dev/lua/nvim-tree/view.lua:81: in function 'wipe_rogue_buffer'
        ...ck/packer/start/nvim-tree.lua.dev/lua/nvim-tree/view.lua:87: in function 'create_buffer'
        ...ck/packer/start/nvim-tree.lua.dev/lua/nvim-tree/view.lua:216: in function 'open'
        ...ack/packer/start/nvim-tree.lua.dev/lua/nvim-tree/lib.lua:78: in function 'open_view_and_draw'
        ...ack/packer/start/nvim-tree.lua.dev/lua/nvim-tree/lib.lua:126: in function 'open'
        ...te/pack/packer/start/nvim-tree.lua.dev/lua/nvim-tree.lua:267: in function 'on_enter'
        ...te/pack/packer/start/nvim-tree.lua.dev/lua/nvim-tree.lua:755: in function <...te/pack/packer/start/nvim-tree.lua.dev/lua/nvim-tree.lua:754>
```

**Comment by Almo7aya** [comment]
@alex-courtis I added a fix for opening a file using :e when the floating window is open, it was opening the file in a new vspilt now it opens the file in a new buffer in the old window, what do u think?

**Comment by alex-courtis** [comment]
> @alex-courtis I added a fix for opening a file using :e when the floating window is open, it was opening the file in a new vspilt, now it opens the file in a new buffer instead of a new window, what do u think?

Let's just keep it consistent with non-floating: open in the previous (not picked) window.

File: lua/nvim-tree.lua [hunk 2]
```diff
@@ -335,6 +335,16 @@ local function setup_autocommands(opts)
   -- reset highlights when colorscheme is changed
   create_nvim_tree_autocmd("ColorScheme", { callback = M.reset_highlight })
 
+  -- prevent new opened file from opening in the same window as nvim-tree
+  create_nvim_tree_autocmd("BufWipeout", {
+    pattern = "NvimTree_*",
+    callback = function()
+      if vim.bo.filetype == "NvimTree" then
```
Comment (alex-courtis) [inline lua/nvim-tree.lua:342]: nice

**Review by alex-courtis — APPROVED** [review]
Tested OK:

- [x] defaults
- [x] quit_on_open
- [x] float
- [x] float quit_on_open

**Pull request merged**